### ISSUE TITLE
Incorrect type. Expected "string" errors after MQTT light effect lists. 

### ISSUE THREAD

**Issue by WayneManion** [keesschollaart81/vscode-home-assistant#817]
After reading 1 & 2 👆, type your issue here and please include the related YAML code like this:

```YAML- name: Dryer Vent Status LED
  platform: mqtt
  command_topic: "Dryer/cmnd/POWER"
  state_topic: "Dryer/tele/STATE"
  payload_on: "ON"
  payload_off: "OFF"
  state_value_template: "{{ value_json.POWER }}"
  availability_topic: "Dryer/tele/LWT"
  payload_available: "Online"
  payload_not_available: "Offline"
  brightness_command_topic: "Dryer/cmnd/Dimmer"
  brightness_state_topic: "Dryer/tele/STATE"
  brightness_scale: 100
  on_command_type: "brightness"
  brightness_value_template: "{{ value_json.Dimmer }}"
  rgb_command_topic: "Dryer/cmnd/Color"
  rgb_state_topic: "Dryer/tele/STATE"
  rgb_value_template: "{{value_json.Color.split(',')[0:3]|join(',')}}"
  color_temp_command_topic: "Dryer/cmnd/CT"
  color_temp_state_topic: "Dryer/tele/STATE"
  color_temp_value_template: "{{ value_json.CT }}"
  effect_command_topic: "Dryer/cmnd/Scheme"
  effect_state_topic: "Dryer/tele/STATE"
  effect_value_template: "{{value_json.Scheme}}"
  effect_list:
      - 0 // this line has red squigglies
      - 1 // this line has red squigglies
      - 2 // this line has red squigglies
      - 3 // this line has red squigglies
      - 4 // this line has red squigglies
 // this line has red squigglies
- name: Nursery Status LED  // THE ERROR MESSAGE APPEARS HERE: Incorrect type. Expected "string"
  platform: mqtt
  command_topic: NurseryTemperature/cmnd/POWER2
  state_topic: NurseryTemperature/tele/STATE
  state_value_template: "{{ value_json.POWER2 }}"
  payload_on: "ON"
  payload_off: "OFF"
  availability_topic: NurseryTemperature/tele/LWT
  payload_available: "Online"
  payload_not_available: "Offline"
  brightness_command_topic: NurseryTemperature/cmnd/Dimmer
  brightness_state_topic: NurseryTemperature/tele/STATE
  brightness_scale: 100
  on_command_type: "brightness"
  brightness_value_template: "{{ value_json.Dimmer }}"
  rgb_command_topic: NurseryTemperature/cmnd/Color2
  rgb_state_topic: NurseryTemperature/tele/STATE
  rgb_value_template: "{{value_json.Color.split(',')[0:3]|join(',')}}"
  retain: false
  qos: 1
```
I have a bunch of similar RGB bulbs and all of them have a similar error on the line after their definition. I've tried re-arranging the device definition (putting other lines after the effect list. I've tried putting the effect list items in quotes. I've tried omitting the dashes. 

The lights work. I can change the colors, the effects, all that stuff in Home Assistant.

**Comment by frenck** [comment]
effect_list is expected to be a string, not a number. While there is magic happening that corrects the error in your YAML, it is actually incorrect.

```yaml
  effect_list:
      - "0"
      - "1"
      - "2"
      - "3"
      - "4"
```

**Comment by WayneManion** [comment]
> effect_list is expected to be a string, not a number. While there is magic happening that corrects the error in your YAML, it is actually incorrect.
> 
> ```yaml
>   effect_list:
>       - "0"
>       - "1"
>       - "2"
>       - "3"
>       - "4"
> ```

That was the first thing I tried, but no luck. I get the same squigglies and error message.

**Comment by frenck** [comment]
That is indeed a bug! Thanks! 

Fixed in #818
